hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientTopicProxy.java

```diff
@@ -77,6 +91,27 @@ public LocalTopicStats getLocalTopicStats() {
         throw new UnsupportedOperationException("Locality is ambiguous for client!");
     }
 
+    @Override
+    public void publishAll(@Nonnull Collection<? extends E> messages) {
+        checkNotNull(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
+        checkNoNullInside(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
+        ClientMessage request = getClientMessage(messages);
+        invokeOnPartition(request);
+    }
+
+    @Override
+    public InternalCompletableFuture<Void> publishAllAsync(@Nonnull Collection<? extends E> messages) {
+        checkNotNull(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
+        checkNoNullInside(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
+        final ClientMessage clientMessage = getClientMessage(messages);
+        return invokeOnPartitionAsync(clientMessage, null);
```

Comment:
ClientMessageDecoder can not be null. 
If you add `completionStage.toCompletableFuture().join();` in your tests you will get NullPointerException. Please add this version to tests as well. 
You can pass ` clientMessage -> null` instead of null to fix the problem easily. 
Please check `publishAsync` method as well for the same.

Reply:
Hello @sancar, I've fixed this and created one more test.